src/ppom-controller.ts

```diff
@@ -431,12 +430,31 @@ export class PPOMController extends BaseControllerV2<
     return !chainStatus[this.#chainId]?.dataFetched;
   }
 
+  /*
+   * Reset data fetched to false for all chains.
+   */
+  #resetDataFetchedForAllChains(): void {
+    this.update((draftState) => {
+      draftState.chainStatus = Object.keys(draftState.chainStatus).reduce(
+        (result, chainId) => {
+          (result as any)[chainId] = {
+            ...draftState.chainStatus[chainId],
+            dataFetched: false,
+          };
+          return result;
+        },
+        {},
+      );
+    });
+  }
+
   /*
    * Update the PPOM configuration for all chainId.
    */
   async #updatePPOM(): Promise<void> {
     const versionInfoUpdated = await this.#updateVersionInfo();
     if (versionInfoUpdated) {
+      this.#resetDataFetchedForAllChains();
```

Comment:
Is it a valid assumption that all chains get updated data in the CDN at the same time? If so, this is good enough. If not, we might want to have a per-chainId mechanism for this.

Reply:
@naugtur : even if we reset this for all chains only those files will be fetched which are new / updated.